src/Simulation/Core/Udts/UDTPartial.cs

```diff
@@ -62,6 +64,9 @@ public ICallable Partial(object partialTuple)
             return (ICallable)Activator.CreateInstance(partialType, new object[] { this, partialTuple });
         }
 
+        /// <inheritdoc/>
+        public RuntimeMetadata? GetRuntimeMetadata(IApplyData args) => throw new NotImplementedException();
```

Comment:
In all other cases, you've returned null when runtime metadata is unavailable; given that, is throwing an exception expected behavior? Will this cause any breakages, for instance, when considering snippets like `Mapped(Complex(_, 0.0), realParts))`?

Reply:
I agree, just return `null` instead?

Reply:
I guess I wasn't sure what this was used for. I implemented partial operations in src/Simulation/Core/Operations/OperationPartial.cs. Is this similar (i.e. do I reuse that logic) or just return `null`?